### ISSUE TITLE
Add new IPCE specific classes to ISIS, part 1/4

### ISSUE THREAD

**Issue by ascbot** [DOI-USGS/ISIS3#2923]
---

Author Name: **Kristin Berry** (Kristin Berry)

Original Assignee: Ian Humphrey

---

~~AbstractProjectItemView~~ ( moved to #4006)
ControlMeasureEditWidget
ControlPointEditWidget
CubeDnView
Footprint2DView

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Ian Humphrey** (Ian Humphrey)
Original Date: 2016-06-15T21:58:09Z

---

Changes are at **/work/users/ihumphrey/m04004/**

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Jesse Mapel** (Jesse Mapel)
Original Date: 2016-06-17T16:38:55Z

---

Everything looks good, just needs history entries for the documentation updates.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Ian Humphrey** (Ian Humphrey)
Original Date: 2016-06-17T19:30:21Z

---

Updated history comments.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Ian Humphrey** (Ian Humphrey)
Original Date: 2016-06-18T00:22:02Z

---

**Currently, the ControlPointEditWidget is used within MosaicControlNetTool in IPCE**. In IPCE, this functionality is not complete, as running qmos in IPCE and opening a control network (which uses the widget's directory to add an instance of the ControlPointEditWidget) causes a segfault (as the directory variable is NULL).

**Once ControlPointEditWidget is in ISIS MosaicControlNetTool in IPCE cannot be merged into ISIS as it will break qmos in ISIS**

Additionally, merging the ControlPointEditWidget class into ISIS would necessitate modifications to either its header file or the make config files, since the QtWidget module is included within this class and the QtWidgets module does not exist in Qt4.8 (it is called the QtGui module in Qt4.8, which is not equivalent to Qt5's QtGui module).

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Jean Backer** (Jean Backer)
Original Date: 2016-06-27T17:19:29Z

---

CubeDnView still not checked into trunk

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Ian Humphrey** (Ian Humphrey)
Original Date: 2016-06-28T22:39:26Z

---

Blocked by #4006 (AbstractProjectItem, which is blocked by BundleSettings and BundleSettingsInfo (need to be merged into trunk first).
